baseapp/baseapp.go

```diff
@@ -507,16 +550,25 @@ func validateBasicTxMsgs(msgs []sdk.Msg) error {
 // Returns the application's deliverState if app is in runTxModeDeliver,
 // otherwise it returns the application's checkstate.
 func (app *BaseApp) getState(mode runTxMode) *state {
-	if mode == runTxModeDeliver {
+	switch mode {
+	case runTxModeDeliver:
 		return app.deliverState
+	case runTxPrepareProposal:
+		return app.prepareProposalState
+	case runTxProcessProposal:
+		return app.processProposalState
+	default:
```

Comment:
nit: should these cases explicitly switch on each of the [mode values](https://github.com/cosmos/cosmos-sdk/blob/61effe82603006a7192cb311787ca8fc776a4461/baseapp/baseapp.go#L28-L33) and have the default `panic` to catch any unhandled modes?